python/ray/data/_internal/planner/plan_all_to_all_op.py

```diff
@@ -17,14 +17,40 @@
 from ray.data._internal.planner.randomize_blocks import generate_randomize_blocks_fn
 from ray.data._internal.planner.repartition import generate_repartition_fn
 from ray.data._internal.planner.sort import generate_sort_fn
-from ray.data.context import DataContext
+from ray.data.context import DataContext, ShuffleStrategy
+
+
+def _plan_hash_shuffle_repartition(
+    data_context: DataContext,
+    logical_op: Repartition,
+    input_physical_op: PhysicalOperator,
+) -> PhysicalOperator:
+    from ray.data._internal.planner.exchange.sort_task_spec import SortKey
+    from ray.data._internal.execution.operators.hash_shuffle import (
+        HashShuffleOperator,
+    )
+
+    normalized_key_columns = SortKey(logical_op._keys).get_columns()
+
+    return HashShuffleOperator(
+        input_physical_op,
+        data_context,
+        key_columns=tuple(normalized_key_columns),  # noqa: type
+        # NOTE: In case number of partitions is not specified, we fall back to
+        #       default min parallelism configured
+        num_partitions=(
+            logical_op._num_outputs or data_context.default_hash_shuffle_parallelism
+        ),
+        should_sort=logical_op._sort,
+        # TODO wire in aggregator args overrides
+    )
 
 
 def plan_all_to_all_op(
     op: AbstractAllToAll,
     physical_children: List[PhysicalOperator],
     data_context: DataContext,
-) -> AllToAllOperator:
+) -> PhysicalOperator:
```

Comment:
We're eventually going to deprecate AlltoAllOperator right?

Reply:
Likely